File: frontend/recetario/lib/data/models/estudiante.dart
class Estudiante {
  final String id;
  final String usuarioId;
  final String? codigoEstudiante;
  final int cicloActual;
  final String? seccion;
  final String? telefono;
  final String? fechaNacimiento;

  Estudiante({
    required this.id,
    required this.usuarioId,
    this.codigoEstudiante,
    required this.cicloActual,
    this.seccion,
    this.telefono,
    this.fechaNacimiento,
  });

  factory Estudiante.fromJson(Map<String, dynamic> json) {
    return Estudiante(
      id: json['id'],
      usuarioId: json['usuario_id'],
      codigoEstudiante: json['codigo_estudiante'],
      cicloActual: json['ciclo_actual'] ?? 1,
      seccion: json['seccion'],
      telefono: json['telefono'],
      fechaNacimiento: json['fecha_nacimiento'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'usuario_id': usuarioId,
      'codigo_estudiante': codigoEstudiante,
      'ciclo_actual': cicloActual,
      'seccion': seccion,
      'telefono': telefono,
      'fecha_nacimiento': fechaNacimiento,
    };
  }
}

File: backend/internal/models/estudiante.go
package models

import (
	"time"
)

type Estudiante struct {
	ID               string    `json:"id"`
	UsuarioID        string    `json:"usuario_id"`
	CodigoEstudiante string    `json:"codigo_estudiante"`
	CicloActual      int       `json:"ciclo_actual"`
	Seccion          string    `json:"seccion,omitempty"`
	Telefono         string    `json:"telefono,omitempty"`
	FechaNacimiento  string    `json:"fecha_nacimiento,omitempty"`
	CreatedAt        time.Time `json:"created_at"`
	UpdatedAt        time.Time `json:"updated_at"`
}

